src/coprocessor/dag/builder.rs

```diff
@@ -85,20 +95,57 @@ impl DAGBuilder {
     ///
     /// The inner-most executor must be a table scan executor or an index scan executor.
     fn build_normal_first_executor<S: Store + 'static>(
+        executor_index: usize,
```

Comment:
Yes. We will introduce `executor_index` for the rest of the executors as well, which will not be zero. To keep interface consistent and readable (i.e. corresponding parameter is the executor_index instead of a `0` that don't know its meaning), I reserved `executor_index` here.